### PR TITLE
Add `CLDVideoPlayer` class

### PR DESCRIPTION
**Pull request by adimiz1**
### Brief Summary of Changes
Add `CLDVideoPlayer` class

#### What does this PR address?
- [ ] GitHub issue (Add reference - #XX)
- [ ] Refactoring
- [x] New feature
- [ ] Bug fix
- [ ] Adds more tests

#### Are tests included?
- [ ] Yes
- [x] No

#### Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I ran the full test suite before pushing the changes and all the tests pass.

### CODE REVIEW

File: src/AdvancedVideo.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import type { CloudinaryVideo } from '@cloudinary/url-gen';
+import { SDKAnalyticsConstants } from './internal/SDKAnalyticsConstants';
+
+interface CLDVideoPlayerProps {
+  videoUrl?: string;
+  cldVideo?: CloudinaryVideo;
+  videoStyle?: StyleProp<ViewStyle>;
+}
+
+interface CLDVideoPlayerMethods {
```
Comment (lbendavid-cloudinary) [inline src/AdvancedVideo.tsx:14]: Can be removed

File: src/AdvancedVideo.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import React, { Component, RefObject } from 'react';
+import { ViewStyle, StyleProp } from 'react-native';
+import { Video } from 'expo-av';
+import 'react-native-url-polyfill/auto';
+import type { CloudinaryVideo } from '@cloudinary/url-gen';
+import { SDKAnalyticsConstants } from './internal/SDKAnalyticsConstants';
+
+interface CLDVideoPlayerProps {
```
Comment (lbendavid-cloudinary) [inline src/AdvancedVideo.tsx:8]:
```suggestion
interface AdvancedVideoProps {
```

File: src/AdvancedVideo.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+interface CLDVideoPlayerMethods {
+  playVideo: () => Promise<void>;
+  pauseVideo: () => Promise<void>;
+}
+
+type VideoRef = Video | null;
+
+const AdvancedVideo = React.forwardRef<VideoRef, CLDVideoPlayerProps>(
```
Comment (lbendavid-cloudinary) [inline src/AdvancedVideo.tsx:21]:
```suggestion
const AdvancedVideo = React.forwardRef<VideoRef, AdvancedVideoProps>(
```

File: src/AdvancedVideo.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  (props, ref) => {
+    const getVideoUri = (): string => {
+      const { videoUrl, cldVideo } = props;
+      if (videoUrl) {
+        return videoUrl;
+      }
+      if (cldVideo) {
+        return cldVideo.toURL({ trackedAnalytics: SDKAnalyticsConstants });
+      }
+      return '';
```
Comment (lbendavid-cloudinary) [inline src/AdvancedVideo.tsx:26]: Maybe we want to show an error or some other informative message for this case?

**Review by lbendavid-cloudinary — APPROVED** [review]

**Review by const-cloudinary — APPROVED** [review]

**Pull request merged**